src/main/drivers/barometer/barometer_bmp388.c

```diff
@@ -0,0 +1,320 @@
+STATIC_UNIT_TESTED int64_t t_lin = 0;
+
+static void bmp388_start_ut(baroDev_t *baro);
+static void bmp388_get_ut(baroDev_t *baro);
+static void bmp388_start_up(baroDev_t *baro);
+static void bmp388_get_up(baroDev_t *baro);
+
+STATIC_UNIT_TESTED void bmp388_calculate(int32_t *pressure, int32_t *temperature);
+
+void bmp388_extiHandler(extiCallbackRec_t* cb)
+{
+#ifdef DEBUG
+    static uint32_t bmp388ExtiCallbackCounter = 0;
+
+    bmp388ExtiCallbackCounter++;
+#endif
+
+    baroDev_t *baro = container_of(cb, baroDev_t, exti);
+
+    uint8_t intStatus = 0;
+    busReadRegisterBuffer(&baro->busdev, BMP388_INT_STATUS_REG, &intStatus, 1);
+}
+
+void bmp388BusInit(busDevice_t *busdev)
+{
+#ifdef USE_BARO_SPI_BMP388
+    if (busdev->bustype == BUSTYPE_SPI) {
+        IOHi(busdev->busdev_u.spi.csnPin); // Disable
+        IOInit(busdev->busdev_u.spi.csnPin, OWNER_BARO_CS, 0);
+        IOConfigGPIO(busdev->busdev_u.spi.csnPin, IOCFG_OUT_PP);
+        spiSetDivisor(busdev->busdev_u.spi.instance, SPI_CLOCK_STANDARD); // XXX
+    }
+#else
+    UNUSED(busdev);
+#endif
+}
+
+void bmp388BusDeinit(busDevice_t *busdev)
+{
+#ifdef USE_BARO_SPI_BMP388
+    if (busdev->bustype == BUSTYPE_SPI) {
+        spiPreinitCsByIO(busdev->busdev_u.spi.csnPin);
+    }
+#else
+    UNUSED(busdev);
+#endif
+}
+
+void bmp388BeginForcedMeasurement(busDevice_t *busdev)
+{
+    // enable pressure measurement, temperature measurement, set power mode and start sampling
+    uint8_t mode = (BMP388_MODE_FORCED << 4) | (1 << 1) | (1 << 0);
+    busWriteRegister(busdev, BMP388_PWR_CTRL_REG, mode);
+}
+
+bool bmp388Detect(const bmp388Config_t *config, baroDev_t *baro)
+{
+    delay(20);
+
+#if defined(BARO_EXTI_PIN)
+    IO_t baroIntIO = IO_NONE;
+#endif
+
+#if defined(BARO_EXTI_PIN) && defined(USE_EXTI)
+    if (config && config->baroInitExtiTag) {
+        baroIntIO = IOGetByTag(config->baroInitExtiTag);
+        IOInit(baroIntIO, OWNER_BARO_EXTI, 0);
+        EXTIHandlerInit(&baro->exti, bmp388_extiHandler);
+        EXTIConfig(baroIntIO, &baro->exti, NVIC_PRIO_BARO_EXTI, IOCFG_IN_FLOATING, EXTI_TRIGGER_RISING);
+        EXTIEnable(baroIntIO, true);
+    }
+#else
+    UNUSED(config);
+#endif
+
+    busDevice_t *busdev = &baro->busdev;
+    bool defaultAddressApplied = false;
+
+    bmp388BusInit(busdev);
+
+    if ((busdev->bustype == BUSTYPE_I2C) && (busdev->busdev_u.i2c.address == 0)) {
+        // Default address for BMP388
+        busdev->busdev_u.i2c.address = BMP388_I2C_ADDR;
+        defaultAddressApplied = true;
+    }
+
+    busReadRegisterBuffer(busdev, BMP388_CHIP_ID_REG, &bmp388_chip_id, 1);
+
+    if (bmp388_chip_id != BMP388_DEFAULT_CHIP_ID) {
+        bmp388BusDeinit(busdev);
+        if (defaultAddressApplied) {
+            busdev->busdev_u.i2c.address = 0;
+        }
+        return false;
+    }
+
+    // read calibration
+    STATIC_ASSERT(sizeof(bmp388_calib_param_t) == BMP388_TRIMMING_DATA_LENGTH, bmp388_calibration_structure_incorrectly_packed);
```

Comment:
Not really sure why this is in here - this should rather go to where `bmp388_calib_param_t` is defined.

Reply:
well I had it there so that it was clear why it was packed, but sure, i'll mve it.